nap/utils.py

```diff
@@ -60,19 +51,21 @@ def flatten_params(k, vs):
 
         # since we can have more than one value for a single key, we use a
         # tuple of two tuples instead of a dictionary
-        params_tuple = tuple(itertools.chain(*flat_params))
-        param_string = urllib.parse.urlencode(params_tuple)
+        params_tuple = tuple(sorted(itertools.chain(*flat_params), key=lambda key_value: key_value[0]))
```

Comment:
I'm not sure what you mean by the first paragraph. This re-ordering of params was explicitly added to improve cache hits/invalidations (https://github.com/CashStar/nap/pull/13). There's some discussion there about how to accomplish that, but that was a concern raised as part of the change, not as a reason for the change.

I do see, however, in that PR's description that the sorting had been taking the value into account but that was reverted to preserve lists but I can't find the history or discussion about why that happened. (I also couldn't find it in https://github.com/CashStar/nap/pull/12 which was the predecessor to 13)